kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/benchmarks/ReadMetadataRunner.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.defaults.benchmarks;
+
+import io.delta.kernel.Scan;
+import io.delta.kernel.ScanBuilder;
+import io.delta.kernel.Snapshot;
+import io.delta.kernel.Table;
+import io.delta.kernel.data.FilteredColumnarBatch;
+import io.delta.kernel.defaults.engine.DefaultEngine;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.utils.CloseableIterator;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.apache.hadoop.conf.Configuration;
+import org.openjdk.jmh.infra.Blackhole;
+
+public class ReadMetadataRunner implements WorkloadRunner {
+  private final Engine engine;
+  private final Scan scan;
+
+  /**
+   * Constructs the ReadMetadataRunner from the workload spec and the base workload directory
+   * containing test tables. This initializes any state necessary to execute this workload as a
+   * benchmark.
+   *
+   * @param baseWorkloadDirPath The base directory containing workload tables.
+   * @param workload The read_metadata workload specification.
+   */
+  public ReadMetadataRunner(Path baseWorkloadDirPath, ReadMetadata workload) {
+    String resolvedTableRoot = getResolvedTableRoot(baseWorkloadDirPath, workload.getTableRoot());
+    engine = DefaultEngine.create(new Configuration());
+    scan = getScanFromWorkloadSpec(engine, workload, resolvedTableRoot);
+  }
+
+  /**
+   * Resolves the table root path based on whether it's a URI or a relative path. If it is a
+   * relative path, it is resolved based on the workload base directory.
+   *
+   * @param baseWorkloadDirPath The base directory containing workload tables.
+   * @param workloadTableRoot The table root path from the workload spec.
+   * @return The resolved table root path.
+   */
+  private static String getResolvedTableRoot(Path baseWorkloadDirPath, String workloadTableRoot) {
```

Comment:
why private static? can we put these private helper methods at the bottom of the class? below any public methods?